components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/factory/LogoutResponseFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.oidc.factory;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.exception.FrameworkException;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityResponse;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.HttpIdentityResponseFactory;
+import org.wso2.carbon.identity.application.authentication.framework.inbound.IdentityResponse;
+import org.wso2.carbon.identity.application.authenticator.oidc.LogoutClientException;
+import org.wso2.carbon.identity.application.authenticator.oidc.LogoutException;
+import org.wso2.carbon.identity.application.authenticator.oidc.LogoutServerException;
+import org.wso2.carbon.identity.application.authenticator.oidc.model.LogoutResponse;
+import org.wso2.carbon.identity.application.authenticator.oidc.util.OIDCErrorConstants;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * This class  builds a HTTP response instance based on the common IdentityRequest format used by
+ * the authentication framework.
+ */
+public class LogoutResponseFactory extends HttpIdentityResponseFactory {
+
+    private static final Log log = LogFactory.getLog(LogoutResponseFactory.class);
+
+    @Override
+    public boolean canHandle(IdentityResponse identityResponse) {
+
+        return (identityResponse instanceof LogoutResponse);
+
+    }
+
+    public boolean canHandle(FrameworkException exception) {
```

Comment:
why no override annotation is here. If it is not an overridden method, add javadoc comments